include/mrdox/Metadata/TemplateParam.hpp

```diff
@@ -29,16 +47,155 @@ namespace mrdox {
 //
 /** A template parameter.
 */
+
+
+struct TemplateParamInfo;
+
+struct TemplateTypeParamInfo
+{
+    // default type for the type template parameter
+    std::optional<TypeInfo> Default;
+};
+
+struct TemplateNonTypeParamInfo
+{
+    // type of the non-type template parameter
+    TypeInfo Type;
+    // non-type template parameter default value (if any)
+    std::optional<llvm::SmallString<16>> Default;
+};
+
+struct TemplateTemplateParamInfo
```

Comment:
I'm getting quite sick of seeing the word template. Maybe we should shorten that to just "T"

Reply:
anything shorter is better, use your best judgement